core/geometry/src/bspline/KnotVector.ts

```diff
@@ -299,35 +326,39 @@ export class KnotVector {
   }
   /** Return fraction of active knot range to knot value. */
   public fractionToKnot(fraction: number): number {
-    fraction = Geometry.clamp(fraction, 0, 1);   // B-splines are not extendable
+    fraction = Geometry.clamp(fraction, 0, 1); // B-splines are not extendable
     return Geometry.interpolate(this.knots[this.degree - 1], fraction, this.knots[this.knots.length - this.degree]);
   }
   /**
    * Evaluate the B-spline basis functions f[] at a parameter u in a knot span.
    * * This method implements the Mansfield-Cox-de Boor recurrence relation.
    * @param knotIndex0 index of the left knot of the span.
    * @param u value in the knot span: knot[knotIndex0] <= u <= knot[knotIndex0 + 1].
-   * @param f preallocated output array of order basis function values
-   * @returns true if and only if output array is sufficiently sized
+   * @param f preallocated output array of order basis function values.
+   * @returns true if and only if output array is sufficiently sized.
    */
   public evaluateBasisFunctions(knotIndex0: number, u: number, f: Float64Array): boolean {
     if (f.length < this.degree + 1)
       return false;
+    const spanIsValid = () => knotIndex0 >= this.degree - 1 && knotIndex0 + this.degree < this.knots.length;
+    const uIsInSpan = () => this.knots[knotIndex0] <= u && u <= this.knots[knotIndex0 + 1];
+    assert(spanIsValid, "knotIndex0 defines a valid knot span");
+    assert(uIsInSpan, "the knot span contains u");
```

Comment:
Shouldn't these four lines (and their copy in `evaluateBasisFunctions1` be pulled out into a private member function?

Reply:
Will make the change